fluss-server/src/test/java/org/apache/fluss/server/RpcServiceBaseTest.java

```diff
@@ -0,0 +1,758 @@
+class RpcServiceBaseTest {
+
+    private ZooKeeperClient mockZkClient;
+    private TablePath testTablePath;
+    private PhysicalTablePath testPartitionPath;
+    private long testTableId;
+    private long testPartitionId;
+    private List<BucketMetadata> testBucketMetadata;
+
+    @BeforeEach
+    void beforeEach() {
+        mockZkClient = mock(ZooKeeperClient.class);
+
+        testTablePath = TablePath.of("testDb", "testTable");
+        testPartitionPath = PhysicalTablePath.of(testTablePath, "partition1");
+        testTableId = 1L;
+        testPartitionId = 100L;
+        testBucketMetadata =
+                Lists.newArrayList(
+                        new BucketMetadata(0, 1, 1, Arrays.asList(1, 2, 3)),
+                        new BucketMetadata(1, 2, 2, Arrays.asList(4, 5, 6)));
+    }
+
+    private Session createTestSession() throws java.net.UnknownHostException {
+        return new Session(
+                (short) 1,
+                "default",
+                false,
+                java.net.InetAddress.getByName("127.0.0.1"),
+                FlussPrincipal.ANY);
+    }
+
+    // A test implementation of RpcServiceBase to access the protected processMetadataRequest method
+    private static class TestRpcServiceBase extends RpcServiceBase {
+        public TestRpcServiceBase() {
+            super(null, null, null, null, null);
+        }
+
+        @Override
+        public void processMetadataRequest(
+                MetadataRequest request,
+                String listenerName,
+                Session session,
+                Authorizer authorizer,
+                ServerMetadataCache metadataCache,
+                MetadataFunctionProvider functionProvider,
+                CompletableFuture<MetadataResponse> responseFuture) {
+            super.processMetadataRequest(
+                    request,
+                    listenerName,
+                    session,
+                    authorizer,
+                    metadataCache,
+                    functionProvider,
+                    responseFuture);
+        }
+
+        @Override
+        public String name() {
+            return "TestRpcService";
+        }
+
+        @Override
+        public void shutdown() {}
+
+        @Override
+        public CompletableFuture<MetadataResponse> metadata(MetadataRequest request) {
+            return null;
+        }
+    }
+
+    @Test
+    void testGetTableMetadataFromZkAsync_Success() throws Exception {
+        // Setup test data
+        Map<Integer, BucketAssignment> bucketAssignments = new HashMap<>();
+        bucketAssignments.put(0, BucketAssignment.of(1, 2, 3));
+        bucketAssignments.put(1, BucketAssignment.of(2, 3, 4));
+        TableAssignment tableAssignment = new TableAssignment(bucketAssignments);
+
+        // Mock ZooKeeper responses
+        when(mockZkClient.getTableAssignment(testTableId)).thenReturn(Optional.of(tableAssignment));
```

Comment:
Please do not use Mockito, it increases maintenance overhead in the long term, which should be avoided. In this case, we can simply register metadata into zk without mocking the zk client.